app.rb

```diff
@@ -1,12 +1,41 @@
+# frozen_string_literal: true
+
 require 'sinatra/base'
+require 'sinatra/flash'
 require 'sinatra/reloader'
+require './lib/user'
+require './database_connection_setup'
 
-class MakersBnb < Sinatra::Base 
+# App class
+class MakersBnb < Sinatra::Base
   configure :development do
     register Sinatra::Reloader
   end
 
+  enable :sessions
+  register Sinatra::Flash
+
   get '/' do
-    'Hello world!'
+    erb :index
+  end
+
+  get '/login' do
+    erb :login
+  end
+
+  post '/user_details' do
+    session[:username] = params[:username]
```

Comment:
are sessions preferable to instance variables? Would be interested to understand if there is a benefit of using sessions here

Reply:
i read that session might slow things down if we had lots of users logged in as would take up memory, prob not an issue for our purposes but worth considering